tests/functional/report_viewer_api/test_report_counting.py

```diff
@@ -0,0 +1,485 @@
+#
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+from collections import Counter
+import logging
+import os
+import unittest
+
+import shared
+from shared.ttypes import ReviewStatus
+from codeCheckerDBAccess.ttypes import ReportFilter
+
+from libtest import env
+
+
+def get_severity_level(name):
+    """ Convert severity name to value. """
+    return shared.ttypes.Severity._NAMES_TO_VALUES[name]
+
+
+def get_filename(path):
+    _, filename = os.path.split(path)
+    return filename
+
+
+class TestReportFilter(unittest.TestCase):
+
+    _ccClient = None
+
+    def setUp(self):
+        test_workspace = os.environ['TEST_WORKSPACE']
+        self.maxDiff = None
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + test_workspace)
+
+        # Get the clang version which is tested.
+        self._clang_to_test = env.clang_to_test()
+
+        self._testproject_data = env.setup_test_proj_cfg(test_workspace)
+        self.assertIsNotNone(self._testproject_data)
+
+        self._cc_client = env.setup_viewer_client(test_workspace)
+        self.assertIsNotNone(self._cc_client)
+
+        # Get the run names which belong to this test.
+        run_names = env.get_run_names(test_workspace)
+
+        runs = self._cc_client.getRunData(None)
+
+        test_runs = [run for run in runs if run.name in run_names]
+        self._runids = [r.runId for r in test_runs]
+
+        self.run1_checkers = \
+            {'core.CallAndMessage': 5,
+             'core.DivideZero': 3,
+             'core.NullDereference': 3,
+             'core.StackAddressEscape': 3,
+             'cplusplus.NewDelete': 5,
+             'deadcode.DeadStores': 5,
+             'unix.Malloc': 1}
+
+        self.run2_checkers = \
+            {'core.CallAndMessage': 5,
+             'core.DivideZero': 3,
+             'core.NullDereference': 3,
+             'cplusplus.NewDelete': 5,
+             'deadcode.DeadStores': 5,
+             'unix.MismatchedDeallocator': 1}
+
+        self.run1_sev_counts = {'30': 1, '20': 5, '40': 19}
+        self.run2_sev_counts = {'30': 1, '20': 5, '40': 16}
+
+        self.run1_files = \
+            {'file_to_be_skipped.cpp': 2,
```

Comment:
In that config there are global project counts. Some of the new test cases operate on multiple runs where the analysis config was changed and the results are different.
I'm not sure the best way is to set run based results in that config file.